mhkit/wave/io/hindcast.py

```diff
@@ -97,4 +135,74 @@ def request_wpto_point_data(data_type, parameter, lat_lon, years, tree=None,
 
             meta = rex_waves.meta.loc[col,:]
             meta = meta.reset_index(drop=True)    
-        return data, meta
\ No newline at end of file
+        return data, meta
+
+def request_wpto_directional_spectrum(lat_lon, year, tree=None, 
+                                 unscale=True, str_decode=True,hsds=True):
+    
+    """ 
+    Returns data from the WPTO wave hindcast hosted on AWS at the specified latitude and longitude point(s), 
+    or the closest available pont(s).
+    Visit https://registry.opendata.aws/wpto-pds-us-wave/ for more information about the dataset and available 
+    locations and years. 
+
+    Note: To access the WPTO hindcast data, you will need to configure h5pyd for data access on HSDS. 
+    Please see the WPTO_hindcast_example notebook for more information.  
+
+    Parameters
+    ----------
+    lat_lon: tuple or list of tuples
+        latitude longitude pairs at which to extract data 
+    year : string 
+        Year to be accessed. The years 1979-2010 available.
+    tree : str | cKDTree (optional)
+        cKDTree or path to .pkl file containing pre-computed tree
+        of lat, lon coordinates, default = None
+    unscale : bool (optional)
+        Boolean flag to automatically unscale variables on extraction
+        Default = True
+    str_decode : bool (optional)
+        Boolean flag to decode the bytestring meta data into normal
+        strings. Setting this to False will speed up the meta data read.
+                Default = True
+    hsds : bool (optional)
+         Boolean flag to use h5pyd to handle .h5 'files' hosted on AWS
+         behind HSDS. Setting to False will indicate to look for files on 
+         local machine, not AWS. Default = True
+
+    Returns
+    ---------
+    data: DataFrame 
+        Data indexed by datetime with columns named for parameter and cooresponding metadata index 
+    meta: DataFrame 
+        location metadata for the requested data location   
+    """
+    assert isinstance(lat_lon, (list,tuple)), 'lat_lon must be of type list or tuple'
+    assert isinstance(year,str), 'years must be a string'
+    assert isinstance(tree,(str,type(None))), 'tree must be a sring'
+    assert isinstance(unscale,bool), 'unscale must be bool type'
+    assert isinstance(str_decode,bool), 'str_decode must be bool type'
+    assert isinstance(hsds,bool), 'hsds must be bool type'
+
+    region = region_selection(lat_lon)
```

Comment:
Is this the best way to throw this error?

Reply:
Is this for the region check? You could do an assert that all regions are equal, but you would need to find the region for each passed lat/lon first. This looks like it will only pass back one region.

Reply:
sorry it looks like that comment was added before my commit got pushed so disregard. Latest commit should have the actual error handling.

Reply:
The method you are using looks good.